### DISCUSSION TITLE
New Source: Shortcut

### DISCUSSION THREAD

**Discussion by kyle-cheung** [airbytehq/airbyte#33152] [New Connector Request]
## Tell us about the new connector you’d like to have
Source: https://shortcut.com/
Destination: Snowflake

## Describe the context around this new connector
Our entire organization uses Shortcut as the project management/workflow app. We would like to be able to dump this data into Snowflake with two current use cases:
- Reverse ETL into Salesforce for CSMs to take action and monitor ongoing product tickets
- Analytics and insights for engineering/product tickets
Once a day sync is enough for us!

## Describe the alternative you are considering or using
N/A

## Are you willing to submit a PR?
Would love to but not technical enough!

**Comment by half0wl** [comment]
I have a use case for this. Is there anyone working on this? If not, could I have this assigned to me please (PR eta early/mid October)?

**Comment by estreeper** [comment]
Hey everyone,

I wouldn't mind picking this up, we also have a customer who wants this integration. However, before putting time toward this I wanted to find out if this is something that Airbyte would want to integrate, since the issue was converted to this discussion and it doesn't appear on the [Community Contributions Connectors board](https://github.com/orgs/airbytehq/projects/50/views/1). We currently have this published as a custom connector to our workspace.

I've looked at other source connectors implemented using the low code CDK, i.e. `airbyte/airbyte-integrations/connectors/source-surveymonkey`, and feel that I have a good idea of what it entails based on those.

There isn't an API endpoint to list all stories in Shortcut, but after talking with their support and digging a bit more I found two things that seem promising:

[Search Stories Old](https://developer.shortcut.com/api/rest/v3#Search-Stories-Old) - suggested by support, has no limitation on number of items returned. It can filter by date ranges so would support incremental sync
[List Group Stories](https://developer.shortcut.com/api/rest/v3#List-Group-Stories) - this is Team-scoped (Group = Team), but has list operations and pagination

The regular [Search Stories](https://developer.shortcut.com/api/rest/v3#Search-Stories) likely would not work, because even though you can search with the query `is:story` to get all of them, it looks like it will throw an error if there are >1000 search results.

Of these options, Search Stories Old seems like it would be the best, because it captures all stories, is the simplest, and was recommended by Shortcut. I used this endpoint in our workspaces with the low-code connector builder.

[a comment by an automated account is not shown]
